tests/test_xopen.py

```diff
@@ -504,3 +514,80 @@ def test_open_many_gzip_writers(tmp_path):
         files.append(f)
     for f in files:
         f.close()
+
+
+def test_pipedcompressionwriter_wrong_mode():
+    with pytest.raises(ValueError) as error:
+        PipedCompressionWriter("test", ["gzip"], "xb")
+    error.match("Mode is 'xb', but it must be")
+
+
+def test_pipedcompressionwriter_wrong_program():
+    with pytest.raises(OSError):
+        PipedCompressionWriter("test", ["XVXCLSKDLA"], "wb")
+
+
+def test_compression_level(tmpdir, gzip_writer):
```

Comment:
We should switch to using `tmp_path` at some point. (Not in this PR.)

Reply:
:+1: Pathlib.Path works great when writing tests.